calculator/src/main/java/edu/nextstep/camp/calculator/CalculatorViewModel.kt

```diff
@@ -47,4 +61,33 @@ class CalculatorViewModel : ViewModel() {
     private fun updateCalculatorText(text: String) {
         _calculatorText.value = text
     }
+
+    fun toggleCalculatorHistoryShowing() {
+        _isShowCalculatorHistory.value = _isShowCalculatorHistory.value?.not()
+    }
+
+    private fun putCalculateHistory(expression: Expression, result: Int) {
+        viewModelScope.launch {
+            calculatorDatabase.calculateHistoryDao().insertCalculateHistory(CalculateHistoryEntity(
+                expression = expression.toString(),
+                result = result,
+            ))
+        }
+    }
+
+    fun getCalculateHistories() {
+        viewModelScope.launch {
+            calculatorDatabase.calculateHistoryDao().getCalculateHistories().collect {
+                _calculateHistories.value = mapToCalculateHistory(it)
+            }
+        }
+    }
```

Comment:
현재 getCalculateHistories 를 Flow 형태로 사용하고 계십니다.
Flow는 Reactive 스트림 형태로 emit 되는 데이터를 받을때 유용한 타입인데요.
사실 여기서는 크게 효용성이 없는것 같긴합니다.

Flow 사용시 API의 chain을 이용하여 좀더 가독성 높은 코드 구성이 가능합니다.
이를 활용하기 위해 Flow를 사용하실수는 있습니다만 오버엔지리어닝 없이 필요한 곳에 필요한 기술을 대입하는것도 
중요한 부분인것 같습니다.

```suggestion
    fun getCalculateHistories() {
        viewModelScope.launch {
            calculatorDatabase.calculateHistoryDao().getCalculateHistories()
                .map(::mapToCalculateHistory)
                .collect(_calculateHistories::setValue)
        }
    }
```

Reply:
맞네요..! 기록 버튼을 클릭할 때마다 history를 가져오면 됐었는데.. 실시간으로 변동되는 상황이 있다고 생각했었습니다 ㅎ;
좋은 리뷰 감사합니다..!